sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/ManagedSchemaTransformProvider.java

```diff
@@ -134,53 +141,62 @@ private Map<String, Object> resolveUnderlyingConfig() {
 
       return YamlUtils.yamlStringToMap(yamlTransformConfig);
     }
+
+    @VisibleForTesting
+    ManagedConfig resolveUnderlyingTransform(PCollectionRowTuple input) {
```

Comment:
I suggest pushing this BQ logic to a ```BQWriteSchemaTransformProvider``` instead of keeping at the ```ManagedSchemaTransformProvider```.

```ManagedSchemaTransformProvider``` will replace ```BigQueryFileLoadsWriteSchemaTransformProvider``` and ```BigQueryStorageWriteApiSchemaTransform``` and will expand to the correct implementation within the "expand" method.

```ManagedSchemaTransformProvider``` will use the ```BigQueryWriteConfiguration``` (which is already shared between the above two implementations).

This way, the Managed layer can be completely abstracted out from the implementation details of the BigQuery write transform.

WDYT ?

Reply:
Makes sense, I like that better.

I moved the logic to BQ side, PTAL!